Forge.TreeWalker/contracts/ForgeSchemaValidationRules.json

```diff
@@ -1,193 +1,281 @@
-      "type": "object",
-      "properties": {
-        "Type": {
-          "type": "string",
-          "enum": [ "Leaf" ]
-        },
-        "Actions": {
-          "patternProperties": {
-            ".*?": {
-              "$ref": "#/definitions/LeafNodeSummaryActionDefinition"
-            }
-          },
-          "minProperties": 1,
-          "maxProperties": 1
-        },
-        "Properties": {
-          "type": "object"
-        }
-      },
-      "additionalProperties": false,
-      "required": [ "Type" ]
-    },
-    "ChildSelectorDefinition": {
-      "type": "object",
-      "properties": {
-        "Label": {
-          "type": "string"
-        },
-        "ShouldSelect": {
-          "type": "string"
-        },
-        "Child": {
-          "type": "string"
-        }
-      },
-      "additionalProperties": false,
-      "required": [ "Label", "Child" ]
-    },
-    "RetryPolicy": {
-      "type": "object",
-      "properties": {
-        "Type": {
-          "type": "string",
-          "enum": [ "None", "FixedInterval", "ExponentialBackoff" ]
-        },
-        "MinBackoffMs": {
-          "type": "number"
-        },
-        "MaxBackoffMs": {
-          "type": "number"
-        }
-      },
-      "required": [ "Type" ]
-    },
-    "ActionDefinition": {
-      "type": "object",
-      "properties": {
-        "Action": {
-          "type": "string"
+        "SelectionTypeNodeDefinition": {
+            "type": "object",
+            "properties": {
+                "Type": {
+                    "type": "string",
+                    "enum": [ "Selection" ]
+                },
+                "ChildSelector": {
+                    "type": "array",
+                    "items": {
+                        "$ref": "#/definitions/ChildSelectorDefinition"
+                    },
+                    "minItems": 1
+                },
+                "Properties": {
+                    "type": "object"
+                }
+            },
+            "additionalProperties": false,
+            "required": [ "Type", "ChildSelector" ]
         },
-        "Input": {
-          "type": "object"
+        "ActionTypeNodeDefinition": {
+            "type": "object",
+            "properties": {
+                "Type": {
+                    "type": "string",
+                    "enum": [ "Action" ]
+                },
+                "Timeout": {
+                    "type": [ "number", "string" ]
+                },
+                "Actions": {
+                    "allOf": [
+                        {
+                            "not": {
+                                "patternProperties": {
+                                    ".*?": {
+                                        "$ref": "#/definitions/SubroutineActionDefinition"
+                                    }
+                                }
+                            }
+                        },
+                        {
```

Comment:
test comment

Reply:
Test reply from codeflow.

---
In reply to: [355064064](https://github.com/microsoft/Forge/pull/6#discussion_r355064064) [](ancestors = 355064064)